### PR TITLE
Catch the base OAuth2Error error, not a custom one

### PR DESCRIPTION
**Pull request by ericholscher**
An update to https://github.com/readthedocs/readthedocs.org/pull/10022

### CODE REVIEW

File: readthedocs/oauth/services/gitlab.py [hunk 1]
```diff
@@ -602,7 +602,7 @@ def send_build_status(self, build, commit, state):
                 'GitLab commit status creation failed.',
                 debug_data=debug_data,
             )
-        except CustomOAuth2Error as e:
+        except OAuth2Error as e:
```
Comment (humitos) [inline readthedocs/oauth/services/gitlab.py:605]: Why not catching directly `InvalidGrantError` and removing the following `if`?

**Review by humitos — COMMENTED** [review]
Looks good, but I think it's better if we catch the _exact_ exception now we know which one is 😄

File: readthedocs/oauth/services/gitlab.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/oauth/services/gitlab.py:605]: That makes more sense, but I figured there was a reason it was implemented this way 🤷

Comment (humitos) [reply readthedocs/oauth/services/gitlab.py]: Nope, the only reason was that I didn't find the correct/exact exception as you did 😄

**Review by humitos — APPROVED** [review]

**Pull request merged**